### PR TITLE
[WIP] Support node agent to watch k8s secrets.

### PR DESCRIPTION
**Pull request by JimmyCYJ**
Support node agent to work as ingress gateway agent, that watches k8s secrets in namespace istio-ingress and returns key/cert in SDS response.

Design doc: https://docs.google.com/document/d/1agtVQADCk32mLwot8iygVGhHcP1Lna7G1Cne4Zwllgg/edit#

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -304,6 +305,24 @@ func (sc *SecretCache) rotate() {
...
+	// If node agent works as ingress gateway agent, searches for kubernetes secret instead of sending
+	// CSR to CA.
+	if sc.fetcher.UseCaClient == false {
+		secretItem, exist := sc.fetcher.FindIngressGatewaySecret(resourceName)
+		if !exist {
+			return nil, fmt.Errorf("cannot find secret %s for ingress gateway", resourceName)
+		}
+
+		return &model.SecretItem{
+			CertificateChain: secretItem.CertificateChain,
```
Comment (quanjielin) [inline security/pkg/nodeagent/cache/secretcache.go]: customer bring cert is only normal key/cert pair, or also include customer root cert ? if also include root cert, more logic is needed

Comment (JimmyCYJ) [reply security/pkg/nodeagent/cache/secretcache.go]: Good question. Per offline discussion with @myidpt , customer does not add root cert. Only  key/cert pairs are used for ingress gateway.

File: security/pkg/nodeagent/cache/secretcache.go [hunk 2]
```diff
@@ -304,6 +305,24 @@ func (sc *SecretCache) rotate() {
...
+			PrivateKey:       secretItem.PrivateKey,
+			ResourceName:     resourceName,
+			Token:            token,
```
Comment (quanjielin) [inline security/pkg/nodeagent/cache/secretcache.go]: in your case token isn't required, you may need to update ```getCredentialToken``` in sdsservice.go to remove that constraint in your case.

**Comment by quanjielin** [comment]
btw master branch is locked now, release-1.1 is still opened for check in

**Comment by JimmyCYJ** [comment]
> btw master branch is locked now, release-1.1 is still opened for check in

I am going to close this PR and create a new one into release-1.1 branch. Thanks.

**Pull request closed without merging**